third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -83,12 +90,26 @@ impl ModuleGenerator {
     pub fn run(
         ctx: &ModuleContext,
         module_env: &ModuleEnv,
-    ) -> (FF::CompiledModule, Option<FF::FunctionHandle>) {
+    ) -> (FF::CompiledModule, SourceMap, Option<FF::FunctionHandle>) {
         let module = move_binary_format::CompiledModule {
-            version: file_format_common::VERSION_6,
+            version: file_format_common::VERSION_EXPERIMENTAL,
             self_module_handle_idx: FF::ModuleHandleIndex(0),
             ..Default::default()
         };
+        let source_map = {
+            let module_name_opt = if module_env.is_script_module() {
+                None
+            } else {
+                let name = IR_AST::ModuleName(IR_SYMBOL::Symbol::from(
+                    ctx.symbol_to_str(module_env.get_name().name()),
+                ));
+                Some(IR_AST::ModuleIdent::new(
+                    name,
+                    module_env.get_name().addr().expect_numerical(),
+                ))
+            };
+            SourceMap::new(ctx.env.to_ir_loc(&module_env.get_loc()), module_name_opt)
+        };
```

Comment:
Any particular reason it's in the surrounding block?  Makes it a little less readable